internal/commands/app/create.go

```diff
@@ -65,19 +65,18 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 	}
 
 	var groupID = cmd.inputs.Project
-	if appRemote.IsZero() {
-		if groupID == "" {
-			id, err := cli.ResolveGroupID(ui, clients.Atlas)
-			if err != nil {
-				return err
-			}
-			groupID = id
+
+	if groupID == "" {
+		id, err := cli.ResolveGroupID(ui, clients.Atlas, appRemote.GroupID)
+		if err != nil {
+			return err
 		}
-	} else {
-		groupID = appRemote.GroupID
+		groupID = id
 	}
 
-	err = cmd.inputs.resolveName(ui, clients.Realm, appRemote)
+	err = cmd.inputs.resolveName(ui, clients.Realm, realm.AppFilter{
+		GroupID: appRemote.GroupID,
+		App:     cmd.inputs.RemoteApp})
```

Comment:
i think we still want to leverage the resolved remote app somehow here.  probably related to my question above about potentially eliminating `remoteApp` as a return value, but just wanted to leave a comment on this line as well

---

[also style nit] keep this line ending with a `,` and move the `)` to its own newline

Reply:
I also thought about adding ClientAppID to the remoteApp struct but I prefer the idea of returning the realm.App from resolveRemoteApp instead as you suggested.